### ISSUE TITLE
"FastChat not installed" issue on /v1/chat/completion call in nvcr.io/nvidia/pytorch:22.12-py3 container.

### ISSUE THREAD

**Issue by AlexanderOllman** [vllm-project/vllm#855]
An interesting one. 

I can run `python -m vllm.entrypoints.openai.api_server --model facebook/opt-125m` without issue, and /completion calls work fine.

However, when I make a /chat/completion call, I receive the following:

`
INFO:     ::1:38566 - "POST /v1/chat/completions HTTP/1.1" 500 Internal Server Error
ERROR:    Exception in ASGI application
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/dist-packages/uvicorn/protocols/http/h11_impl.py", line 408, in run_asgi
    result = await app(  # type: ignore[func-returns-value]
  File "/usr/local/lib/python3.8/dist-packages/uvicorn/middleware/proxy_headers.py", line 84, in __call__
    return await self.app(scope, receive, send)
  File "/usr/local/lib/python3.8/dist-packages/fastapi/applications.py", line 289, in __call__
    await super().__call__(scope, receive, send)
  File "/usr/local/lib/python3.8/dist-packages/starlette/applications.py", line 122, in __call__
    await self.middleware_stack(scope, receive, send)
  File "/usr/local/lib/python3.8/dist-packages/starlette/middleware/errors.py", line 184, in __call__
    raise exc
  File "/usr/local/lib/python3.8/dist-packages/starlette/middleware/errors.py", line 162, in __call__
    await self.app(scope, receive, _send)
  File "/usr/local/lib/python3.8/dist-packages/starlette/middleware/cors.py", line 83, in __call__
    await self.app(scope, receive, send)
  File "/usr/local/lib/python3.8/dist-packages/starlette/middleware/exceptions.py", line 79, in __call__
    raise exc
  File "/usr/local/lib/python3.8/dist-packages/starlette/middleware/exceptions.py", line 68, in __call__
    await self.app(scope, receive, sender)
  File "/usr/local/lib/python3.8/dist-packages/fastapi/middleware/asyncexitstack.py", line 20, in __call__
    raise e
  File "/usr/local/lib/python3.8/dist-packages/fastapi/middleware/asyncexitstack.py", line 17, in __call__
    await self.app(scope, receive, send)
  File "/usr/local/lib/python3.8/dist-packages/starlette/routing.py", line 718, in __call__
    await route.handle(scope, receive, send)
  File "/usr/local/lib/python3.8/dist-packages/starlette/routing.py", line 276, in handle
    await self.app(scope, receive, send)
  File "/usr/local/lib/python3.8/dist-packages/starlette/routing.py", line 66, in app
    response = await func(request)
  File "/usr/local/lib/python3.8/dist-packages/fastapi/routing.py", line 273, in app
    raw_response = await run_endpoint_function(
  File "/usr/local/lib/python3.8/dist-packages/fastapi/routing.py", line 190, in run_endpoint_function
    return await dependant.call(**values)
  File "/root/vllm/vllm/entrypoints/openai/api_server.py", line 203, in create_chat_completion
    prompt = await get_gen_prompt(request)
  File "/root/vllm/vllm/entrypoints/openai/api_server.py", line 73, in get_gen_prompt
    raise ModuleNotFoundError(
ModuleNotFoundError: fastchat is not installed. Please install fastchat to use the chat completion and conversation APIs: '$ pip install fschat'
`

`pip show fschat` returns:

`Name: fschat
Version: 0.2.24
Summary: An open platform for training, serving, and evaluating large language model based chatbots.
Home-page:
Author:
Author-email:
License:
Location: /usr/local/lib/python3.8/dist-packages
Requires: fastapi, httpx, markdown2, nh3, numpy, prompt-toolkit, pydantic, requests, rich, shortuuid, tiktoken, uvicorn
Required-by:
`

transformers version is also 4.32.0.

**Comment by nepalprabin** [comment]
Try downgrading fastchat:
`pip install fschat==0.2.23`

However, this may raise another issue. I am getting tokenizer error..

**Comment by AlexanderOllman** [comment]
Sadly, no dice. I receive the same error following a downgrade of fschat. 

To confirm the downgrade:

```
root@localhost:~/vllm# pip show fschat
Name: fschat
Version: 0.2.23
Summary: An open platform for training, serving, and evaluating large language model based chatbots.
Home-page:
Author:
Author-email:
License:
Location: /usr/local/lib/python3.8/dist-packages
Requires: accelerate, einops, fastapi, gradio, httpx, markdown2, nh3, numpy, peft, prompt-toolkit, pydantic, requests, rich, sentencepiece, shortuuid, tiktoken, tokenizers, torch, transformers, uvicorn, wandb
Required-by:
```

**Comment by zattrorc** [comment]
Pip uninstall transformer-engine
Solve my problem

**Comment by AlexanderOllman** [comment]
> 

`transformer-engine` is not installed. Should it be?

**Comment by AlexanderOllman** [comment]
Update: this issue was resolved following a clean install of the Ubuntu 22.04 image from docker hub. 

It **did** end up requiring `fschat==0.2.23`. Any higher threw the same error.

Here is what I run to get start the container and get it to a state where it can run on multiple GPUs. Working fine for everything but the /v1/chat/completion call:

```
podman run --interactive --tty --network=host --name test_llm --shm-size 10.24g --device nvidia.com/gpu=all ubuntu
apt-get update
apt-get install wget nano python3 python3-pip git 
wget https://developer.download.nvidia.com/compute/cuda/repos/ubuntu2204/x86_64/cuda-keyring_1.1-1_all.deb
dpkg -i cuda-keyring_1.1-1_all.deb
apt-get update
apt-get install cuda-11-8
pip3 install vllm
pip3 uninstall torch
pip3 install pandas ray fschat
pip3 install fschat==0.2.23
pip3 install torch torchvision torchaudio --index-url https://download.pytorch.org/whl/cu118
```

Then running the API server commands with `tensor-parallel-size 4` appears to work nicely.

**Comment by xhluca** [comment]
I ran into this issue (regular python install), I just tried to `pip install accelerate` and it solved it

**Comment by james-batchelor** [comment]
The error is getting raised from here https://github.com/vllm-project/vllm/blob/main/vllm/entrypoints/openai/api_server.py#L74
```    if not _fastchat_available:
        raise ModuleNotFoundError(
            "fastchat is not installed. Please install fastchat to use "
            "the chat completion and conversation APIs: `$ pip install fschat`"
        )
```
Which comes from
```try:
    import fastchat
    from fastchat.conversation import Conversation, SeparatorStyle
    from fastchat.model.model_adapter import get_conversation_template
    _fastchat_available = True
except ImportError:
    _fastchat_available = False
```
So I recommend trying to import these lines in your python shell and fixing things from there. In my case the imports went through `accelerate` and errored on `import transformer_engine_extensions as tex`. Uninstalling `transformer-engine` as suggested above solved the issue for me (I am using `fschat==0.2.23` and `accelerate==0.23.0`).

**Comment by mclamee** [comment]
> I ran into this issue (regular python install), I just tried to `pip install accelerate` and it solved it

I met the same issue and this solution works perfectly for me

**Comment by kyritzb** [comment]
> I ran into this issue (regular python install), I just tried to `pip install accelerate` and it solved it

That helped! Thank you :)
